### PR TITLE
Do not always start with reload option (fixes #490)

### LINKED ISSUE #490: kinto start takes too much time.
It can sometimes take more than 2 seconds, as reported by @n1k0 on JS clients integration tests.

I investigated a bit and found out, that on my machine, loading the entrypoint takes more than 1 second already:
```
$ time python -c "from pkg_resources import load_entry_point; load_entry_point('kinto', 'console_scripts', 'kinto')"
python -c   0,96s user 0,16s system 99% cpu 1,132 total
```

In comparison, `pserve` takes 200msec: 
```
$ time python -c "from pkg_resources import load_entry_point; load_entry_point('pyramid', 'console_scripts', 'pcreate')"
python -c   0,18s user 0,09s system 98% cpu 0,272 total
```

I realized that moving `import requests` from `cliquet.initialization` imports [PR](https://github.com/mozilla-services/cliquet/pull/674), and remove `import pip` from `kinto.__main__` I could reduce by half:
```
$ time python -c "from pkg_resources import load_entry_point; load_entry_point('kinto', 'console_scripts', 'kinto')"
python -c   0,36s user 0,18s system 98% cpu 0,543 total
```

I knew this was not going to speed up the `kinto start` command too much. I tracked down and noticed the `__main__:main` was executed twice because of `--reload` argument.

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

**Comment by leplatrem** [comment]
@almet r?

**Comment by Natim** [comment]
This looks great :+1:

File: kinto/__main__.py [hunk 1]
```diff
@@ -45,6 +45,11 @@ def main(args=None):
     parser_migrate.set_defaults(which='migrate')
 
     parser_start = subparsers.add_parser('start')
+    parser_start.add_argument('--reload',
+                              action='store_true',
+                              help='Reload when code or config changes',
```
Comment (almet) [inline kinto/__main__.py]: consider: "Restart the kinto server on code or configuration changes"

**Comment by almet** [comment]
Thanks !

**Pull request merged**